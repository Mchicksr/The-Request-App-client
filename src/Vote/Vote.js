import React from 'react';
import Counter from './counter'
import './Vote.css'


class Vote extends React.Component {
    
   
    render(){
       
        const capitalize = (str) =>{
            if(str)
            return str.charAt(0).toUpperCase() + str.slice(1)
        } 

        const  songs  = this.props.songs.map((songs, name,title,artist) =>
            
                <ul>
                    <li key={name} >Name: {capitalize(songs.name)}</li>
                    <li key={title}>Title: {capitalize(songs.title)}</li>
                    <li key={artist}>Artist: {capitalize(songs.artist)}</li>
                    <Counter />
                    
                </ul>
                        
                        )
                    
               
       
        
    return(
       

        <div>
             <p>Check out what songs your fellow party-goers have submitted and vote your favorite ones as many times as you want! You can even vote for your own submission! The song with the highest votes may be the one that is played next!</p>
             
            <h1 className="neon">VOTE</h1><br></br>
            <h1 className="songs">{songs}</h1>
            
            
        </div>
    )
    }
}

export default Vote; 

