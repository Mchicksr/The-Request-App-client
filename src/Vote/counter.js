import React from 'react';

export default class Counter extends React.Component{
    constructor(props){
        super(props) 
        // debugger
        // console.log('init1')
        this.state ={
            // FavSong:this.props.songs
            FavSong:[ 
                {className: 'name', votes: 0},
                
            
        ]
        }
        
    }

    vote(i){
        
        let VoteSong = [...this.state.FavSong]
        VoteSong[i].votes++
        this.setState({FavSong: VoteSong});
    }


    // async handleVote(e){
    //     e.preventDefault()
        
    //     const {votes}= this.state;
    //     const findVote = {votes}
    //     const get={
    //         method:"GET",
    //         body:JSON.stringify(vote),
    //         headers:{
    //             'content-Type': 'application/json'
    //         }
           

    //     }
    //     const {params} = this.props.match;
    //     fetch(`https://mighty-temple-37477.herokuapp.com/api/songs/`)
    //     .then((res)=>{
    //         if(!res.ok){
    //             return res.json().then(e=> Promise.reject(e))
    //         }
    //         return res.json()
    //     })

    
    //     .then((res)=>{
    //         this.setState({vote:res})
    //         console.log('TEST2',res)
    //     })
    //     .catch(error=>{
    //         console.log(error)
    //     })
    //     console.log("TEST1!", this.props.songs)
    // }

render(){
    return(
        // <h1>hey</h1>
        <div>
             {this.state.FavSong.map((songV,i)=>
                <div key={i} className= "song"  >
                        <div className ="songVote">
                            {songV.votes}
                        </div>
                    
                    <button 
                            onClick={this.vote.bind(this,i,songV)}
                            // onChange={this.handleVote}
                            >vote
                          
                    </button>
                </div>
             )}            
        </div>
    )
}
} 