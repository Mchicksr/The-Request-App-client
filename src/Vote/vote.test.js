import React from 'react'
import Vote from './Vote'
import ReactDOM from 'react-dom'

import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({adapter:new Adapter()})
import renderer from 'react-test-renderer' 


it('renders without crashingV,',()=>{
    const div = document.createElement('div');
    ReactDOM.render( <Vote songs={[]}/>, div)
    })


    it("matches snapshot main", () => {
        // const tree = renderer.create(<Button label="click me please"></Button>).toJSON();
          const tree = renderer.create(  <a class="req" href="/Main"> Send Request </a>).toJSON();
      
        expect(tree).toMatchSnapshot();
      })

