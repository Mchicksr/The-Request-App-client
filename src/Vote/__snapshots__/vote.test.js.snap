// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot main 1`] = `
<a
  class="req"
  href="/Main"
>
   Send Request 
</a>
`;
